sdk/kubeflow/trainer/api/trainer_client.py

```diff
@@ -119,21 +119,24 @@ def list_runtimes(self) -> List[types.Runtime]:
                                 if container.name == constants.CONTAINER_TRAINER:
                                     resources = container.resources
 
-                    # TODO (andreyvelich): Currently, only Torch is supported for NumProcPerNode.
-                    num_procs = (
-                        ml_policy.torch.num_proc_per_node if ml_policy.torch else None
-                    )
-
                     # Get the accelerator for the Trainer nodes.
                     # TODO (andreyvelich): Currently, we get the accelerator type from
                     # the runtime labels.
-                    _, accelerator_count = utils.get_container_devices(
-                        resources, num_procs
-                    )
-                    if accelerator_count != constants.UNKNOWN:
-                        accelerator_count = str(
-                            int(accelerator_count) * int(ml_policy.num_nodes)  # type: ignore
-                        )
+                    _, accelerator_count = utils.get_container_devices(resources)
+
+                    # NumProcPerNode from Torch or MPI overrides accelerator count.
```

Comment:
This is just a comment.

Ideally, we want to store the final deployed job specifications like commands, image, and numNodes in TrainJob Status. I'm currently thinking design. After that SDK can delegate such calculation to TrainJob.

Reply:
Since TrainJob values are immutable.

Reply:
Oh, that is good point!

Reply:
In any case, before we start such status design, we need to finalize PodSpecOverride design so that we can identify which files should be stored in status.

Anyway, this SDK side calculation is still valid here